src/controllers/productController.js

```diff
@@ -0,0 +1,51 @@
+import { product } from '../database/models';
+
+const getProductById = async (req, res) => {
+  console.log('hello');
+  try {
+    const { id } = req.params;
+    const specificItem = await product.findOne({
+      where: { id },
+    });
+
+    if (!id) {
+      return res.status(404).json({
+        status: 'fail',
+        message: 'Missing id parameter in request.',
+      });
+    }
+
+    if (!specificItem) {
+      return res.status(404).json({
+        status: 'fail',
+        message: 'Product not found.',
+      });
+    }
+
+    if (!specificItem.stock === 'In Stock') {
+      return res.status(404).json({
+        status: 'fail',
+        message: 'Product not available for sale.',
+      });
+    }
+
+    if (req.user && req.user.role.roleName === 'user' && req.user.id !== specificItem.userId) {
+      return res.status(403).json({
+        status: 'fail',
+        message: 'You are not allowed to perform this operation',
+      });
+    }
+
+    res.status(200).json({
+      status: 'success',
+      item: specificItem,
+    });
+  } catch (err) {
+    res.status(400).json({
+      status: 'fail',
+      message: err.message,
+    });
+  }
+};
+
```

Comment:
I think another endpoint specifically for when a seller wants to retrieve a specific item from their own collection would be needed.